docs/DEPLOYMENT.md

```diff
@@ -0,0 +1,410 @@
+    ```bash
+    git clone https://github.com/macports/macports-webapp.git
+    cd macports-webapp
+    ```
+ - Build the docker image
+    ```bash
+    docker build -t macports-webapp .
+    ```
+ - Once build finishes, the image is ready to run.
+ 
+### 1.b: Downloading image from Docker registry
+
+This is rather simple, but useless if your purpose is development as you won't be able to test your changes.
+
+```bash
+docker pull arjunsalyan/macports-webapp
+```
+
+### 2. Starting the container
+
+After using any one of the methods described in (1.a) or (1.b), you have an image with you. We will start this image with
+several environment variables that are needed to connect to PostgreSQL, Solr and memcached. A file `env.sample` has been
+added in the repo for your convenience.
+
+```bash
+DB_NAME=                            (name of our database, webapp)
+DB_USER=                            (username of the database user)
+DB_PASSWORD=                        (password for your database user)
+DB_HOST=                            (localhost)
+DB_PORT=                            (default is 5432)
+SECRET_KEY=                         (any string, but secret)
+SOLR_URL=                           (http://127.0.0.1:8983/solr/tester)
+SOLR_ADMIN_URL=                     (http://127.0.0.1:8983/solr/admin/cores)
+EMAIL_HOST=                         (smtp host, optional)
+EMAIL_PORT=                         (smtp port, optional)
+EMAIL_HOST_USER=                    (smtp user, optional)
+EMAIL_HOST_PASSWORD=                (smtp user password, optional)
+```
+
+On a mac, `host.docker.internal` should be used instead of "localhost" or "127.0.0.1".
+
+Set values for the variables and save this as a file, let's name it `env`.
+
+Environment variables are ready, we will also mount some volumes to our Docker container. Get the path to your solr 
+directory, from the step where we installed Solr. Also, create a folder for the app to store some utility data 
+(portindex, portindex.json etc.). We will mount both these directories as volumes to the docker container, call them `path/to/solr`
+and `path/to/data`.
+
+Start the image:
+
+On Linux:
+```bash 
+docker run -d \
+    -v /path/to/solr:/solr \
+    -v /path/to/data:/code/app/data \
+    --network="host" \
+    --name=webapp-container \
+    --env-file=env \
+    macports-webapp
+```
+
+Instead of "macports-webapp" user "arjunsalyan/macports-webapp" as the image name if you pulled the image from
+docker registry.
+
+On Mac:
+```bash
+sudo docker run -d \
+    -p 8080:8080 \
+    -v /path/to/solr:/solr \
+    -v /path/to/data:/code/app/data \ 
+    --name=webapp-container \
+    --env-file=env \ 
+    macports-webapp
+```
+
+Open `127.0.0.1:8080` in the browser to test if the app has started, you won't have any ports though.
+
+### 3. Setting up initial data
+These instructions are common for the docker and without docker, so please skip the next section and jump to
+"Initialising the web app"
```

Comment:
This section sounds relatively confusing to me.

You could add a sentence (without its own section) like:
```
Continue with [Initialising the Webapp](#initialising-the-webapp).
```